File: examples/canonical-auth/ruby/lib/tenantcore_auth/auth/auth_context.rb
module TenantcoreAuth
  module Auth
    AuthContext = Struct.new(
      :sub, :email, :tenant_id, :tenant_role,
      :groups, :permissions, :acl_ver,
      :issued_at, :expires_at,
      keyword_init: true
    ) do
      def has_permission?(permission)
        permissions.include?(permission)
      end
    end
  end
end

File: examples/canonical-auth/elixir/lib/tenantcore_auth/auth/auth_context.ex
defmodule TenantcoreAuth.Auth.AuthContext do
  defstruct [:sub, :email, :tenant_id, :tenant_role, :groups, :permissions, :acl_ver, :issued_at, :expires_at]

  def has_permission?(%__MODULE__{permissions: permissions}, permission) do
    permission in permissions
  end
end
